geochem_inverse_optimize.py

```diff
@@ -281,6 +280,279 @@ def get_roughness(self) -> float:
         return cp.norm(cp.vstack(self._regularizer_terms)).value
 
 
+class SampleNetworkContinuous:
+
+    # TODO: Docstrings
+
+    def __init__(self, sample_network: nx.DiGraph, area_labels: np.array, nx: int, ny: int) -> None:
+        self.sample_network = sample_network
+        self.grid = InverseGrid(nx, ny, area_labels, sample_network)
+        self._site_to_parameter: Dict[str, ReciprocalParameter] = {}
+        self._primary_terms = []
+        self._regularizer_terms = self._build_regularizer_terms()
+        self._regularizer_strength = cp.Parameter(nonneg=True)
+        self._problem = None
+        self._build_primary_terms()
+        self._build_problem()
+
+    def _build_primary_terms(self) -> None:
+        for _, data in self.sample_network.nodes(data=True):
+            data["data"].total_flux = 0.0
+
+        # Build the main objective
+        # Use a topological sort to ensure an upstream-to-downstream traversal
+
+        # TODO: Delete this legend when labels.tif is updated
+        # Set up a number <-> string legend for upstream areas
+
+        for sample_name, my_data in nx_topological_sort_with_data(self.sample_network):
+            # # TODO: Update this to work with new labels.tif output
+            concs = [
```

Comment:
It grabs all the `InverseNode`s associated with a particular sample site and each `InverseNode` corresponds to a rectangle of pixels on the base raster. 

The way we associate `InverseNode`s to sample sites is based on sub-catchment that the *centre* of that rectangle lies in. Note that for rectangles which overlap between two sub-catchments, this will introduces some inaccuracies as this system means each `InverseNode` can be associated with just one catchment, whereas in reality a node may contribute to both. However, as the resolution increases the magnitude of this error is decreased. We could construct a more complex system that allows for `InverseNode`s to donate to multiple subcatchments but I don't think this is a major error at this present time and it would be a bit of a pain.

Reply:
You may wish to add comments explaining this somewhere.

Reply:
Added to doc string of `InverseNode` class